godot-core/src/builtin/quaternion.rs

```diff
@@ -30,10 +30,14 @@ impl Quaternion {
         Self { x, y, z, w }
     }
 
+    /// Creates a Quaternion from a Vector3 and an angle.
+    ///
+    /// # Panics
+    /// If the Vector3 is not normalized.
     pub fn from_angle_axis(axis: Vector3, angle: real) -> Self {
         let d = axis.length();
-        if d == 0.0 {
-            Self::new(0.0, 0.0, 0.0, 0.0)
+        if d != 1.0 {
```

Comment:
Checking for exact equality with floating points is not a good idea; furthermore, we already have a method for this check -- you can use `Vector3::is_normalized()` here.